src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,845 @@
+# -*- coding: utf-8 -*-
+'''
+Created on 19 June 2018
+
+@author: Anders Muskens
+
+Copyright © 2018 Anders Muskens, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+from __future__ import division
+from concurrent.futures import CancelledError, TimeoutError
+import fcntl
+import glob
+import serial
+import logging
+import os
+import threading
+import time
+import copy
+import re
+from Pyro4.core import isasync
+from odemis import model
+from odemis.util import driver
+from odemis.model import HwError, CancellableFuture, CancellableThreadPoolExecutor
+
+# Unit definitions 
+UNIT_DEF = {
+    'mm': 2,
+    'um': 3,
+    'in': 4,
+    }
+
+
+class ESPError(model.HwError):
+    """
+    Exception used to indicate a problem reported by the device.
+    """
+    pass
+
+
+class ESP(model.Actuator):
+
+    def __init__(self, name, role, port, axes=None, inverted=None, **kwargs):
+        """
+        A driver for a Newport ESP 301 Stage Actuator. This driver supports a serial or USB
+        connection to a Unix host.
+
+        name: (str)
+        role: (str)
+        port: (str) port name. Can be a pattern, in which case all the ports
+          fitting the pattern will be tried.
+          Use /dev/fake for a simulator
+        axes: dict str (axis name) -> dict (axis parameters)
+            axis parameters: {
+                number: (int) either 1,2,3
+                range: [float, float]
+                conv_factor (float): a conversion factor that converts to the devices internal unit (mm)
+                    By default, the position VA is displayed as metres, so this value takes 1000.
+            }
+
+        inverted: (bool) defines if the axes are inverted
+
+        The offset can be specified by setting MD_POS_COR as a coordinate dictionary
+
+        """
+
+        if len(axes) == 0:
+            raise ValueError("Needs at least 1 axis.")
+
+        # Connect to serial port
+        self._ser_access = threading.Lock()
+        self._serial = None
+        self._file = None
+        self._port, self._version = self._findDevice(port)  # sets ._serial and ._file
+        logging.info("Found Newport ESP301 device on port %s, Ver: %s",
+                     self._port, self._version)
+
+        self._offset = {}
+        self._axis_conv_factor = {}
+        self._axis_unit = {}
```

Comment:
Drop this `_axis_unit`, as it's unused (excepted for the line that _writes_ to it).